pkg/core/config/main.go

```diff
@@ -242,26 +243,25 @@ func (config *Config) Update(data interface{}) (err error) {
 				Retrieve the value of a third location key from
 				the updatecli contex.
 				It returns an error if a key doesn't exist
-				It returns {{ pipeline "<key>" }} if a key exist but still set to zero value,
+				It returns {{ source "<key>" }} if a key exist but still set to zero value,
 				then we assume that the value will be set later in the run.
 				Otherwise it returns the value.
 				This func is design to constantly reevaluate if a configuration changed
 			*/
 
-			val, err := getFieldValueByQuery(
-				data, []string{"Sources", s, "Output"})
-
+			sourceResult, err := getFieldValueByQuery(data, []string{"Sources", s, "Result"})
 			if err != nil {
 				return "", err
 			}
 
-			if len(val) > 0 {
-				return val, nil
+			switch sourceResult {
+			case result.SUCCESS:
+				return getFieldValueByQuery(data, []string{"Sources", s, "Output"})
+			case result.FAILURE:
+				return "", fmt.Errorf("Parent source %q failed", s)
+			default:
+				return fmt.Sprintf("{{ source %q }}", s), nil
 			}
-			// If we couldn't find a value, then we return the function so we can retry
```

Comment:
@dduportal Any reason to have to drop the comment that explains the default value?

Reply:
Fixed, I've added a comment that explain the "default" case WDYT?